app/controllers/web_rake/tasks_controller.rb

```diff
@@ -80,81 +79,45 @@ def ensure_environment_loaded
-      custom_task_names = extract_task_names_from_files
-
-      # Always include db:seed task
-      custom_task_names << 'db:seed'
-
       # Ensure all tasks are loaded
       Rails.application.load_tasks
 
-      # Get only the tasks that match our custom task names
-      custom_task_names.filter_map do |name|
-        Rake::Task[name] rescue nil
-      end.select(&:actions)
-    end
-
-    def extract_task_names_from_files
-      task_names = []
-
-      # Parse each rake file to extract task names
-      Dir.glob(Rails.root.join('lib/tasks/**/*.rake')).each do |file|
-        content = File.read(file)
-
-        # Match various task definition patterns
+      tasks = []
 
-        # Pattern for: task({ :sample_data => :environment })
-        content.scan(/task\s*\(\s*\{\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task({ sample_data: :environment }) - Ruby 1.9+ syntax with parens and braces
-        content.scan(/task\s*\(\s*\{\s*([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
+      # Find tasks defined in lib/tasks or db:seed
+      Rake.application.tasks.each do |task|
+        next if task.actions.blank?
 
-        # Pattern for: task :sample_data => :environment
-        content.scan(/task\s+:([a-zA-Z_][a-zA-Z0-9_]*)\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task "sample_data" => :environment
-        content.scan(/task\s+["']([a-zA-Z_][a-zA-Z0-9_]*)["']\s*=>/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task sample_data: :environment (Ruby 1.9+ syntax)
-        content.scan(/task\s+([a-zA-Z_][a-zA-Z0-9_]*)\s*:/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: task(:sample_data)
-        content.scan(/task\s*\(\s*:([a-zA-Z_][a-zA-Z0-9_]*)\s*\)/).each do |match|
-          task_names << match[0]
-        end
-
-        # Pattern for: desc "..." \n task :name
-        content.scan(/desc\s+.*?\n\s*task\s+:([a-zA-Z_][a-zA-Z0-9_]*)/).each do |match|
-          task_names << match[0]
+        if task_from_lib_tasks?(task) || task.name == "db:seed"
+          tasks << task
         end
       end
 
-      task_names.uniq
+      tasks
     end
 
     def find_task
       task_name = params[:id].gsub('__', ':')
 
-      # Check if this is one of our allowed tasks (custom tasks + db:seed)
-      allowed_tasks = extract_task_names_from_files + ['db:seed']
-      return nil unless allowed_tasks.include?(task_name)
-
       # Ensure tasks are loaded
       Rails.application.load_tasks
 
       # Find the task
-      Rake::Task[task_name] rescue nil
+      task = Rake::Task[task_name] rescue nil
+      return nil unless task
+
+      # Check if it's db:seed or defined in lib/tasks
+      return task if task_name == 'db:seed'
+      return task if task_from_lib_tasks?(task)
```

Comment:
You could make this one if statement with an or operand to be consistent with the previous change 

```suggestion
      return task if task_from_lib_tasks?(task) || task_name == "db:seed"
```